src/bidiMapper/domains/context/BrowsingContextImpl.ts

```diff
@@ -848,11 +832,161 @@ export class BrowsingContextImpl {
     }
   }
 
+  async #parseRect(clip?: BrowsingContext.ClipRectangle) {
+    if (!clip) {
+      return;
+    }
+    switch (clip.type) {
+      case 'viewport':
+        return {x: clip.x, y: clip.y, width: clip.width, height: clip.height};
+      case 'element': {
+        if (clip.scrollIntoView) {
+          throw new UnsupportedOperationException(
+            `'scrollIntoView' is currently not supported`
+          );
+        }
+        const sandbox = await this.getOrCreateSandbox(undefined);
+        const result = await sandbox.callFunction(
+          String((element: unknown) => {
+            return element instanceof Element;
+          }),
+          {type: 'undefined'},
+          [clip.element],
+          false,
+          Script.ResultOwnership.None,
+          {}
+        );
+        if (result.type === 'exception') {
+          throw new NoSuchElementException(
+            `Element ${clip.element.sharedId} was not found`
+          );
+        }
+        assert(result.result.type === 'boolean');
+        if (!result.result.value) {
+          throw new NoSuchElementException(
+            `Node ${clip.element.sharedId} is not an Element`
+          );
+        }
+        {
+          const result = await sandbox.callFunction(
+            String((element: Element) => {
+              const rect = element.getBoundingClientRect();
+              return {
+                x: rect.x,
+                y: rect.y,
+                height: rect.height,
+                width: rect.width,
+              };
+            }),
+            {type: 'undefined'},
+            [clip.element],
+            false,
+            Script.ResultOwnership.None,
+            {}
+          );
+          assert(result.type === 'success');
```

Comment:
Would this crash the mapper if it's false?

Is it better to throw an Exception instead?

(I am not sure, just asking.)

Reply:
Given this is in a command it should be caught and turned into `Unknown Exeption`, if you looks at assert's implementation it checks the value and throws an error.

Reply:
@Lightning00Blade did you mean this as an AI or just a statement?

Reply:
From what I remember when we try to process a command we do it though a `try/catch` if anything fails and the error is not of Type `Exeption` (Internal class of all BiDi specific errors) we return a Unknown error.